railties/lib/rails/generators/rails/resource_route/resource_route_generator.rb

```diff
@@ -21,7 +25,9 @@ def add_resource_route
         end
 
         # inserts the primary resource
-        write("resources :#{file_name.pluralize}", route_length + 1)
+        resource_string = "resources :#{file_name.pluralize}"
+        resource_string << ", except: [:new, :edit]" if options.api?
```

Comment:
Yeah, I'm curious how `resources` generate only 5 routes in a API-only apps because it seems to be causing https://github.com/rails/rails/issues/24264